### PR TITLE
User picture icon support

### PR DESCRIPTION
**Pull request by gizzmojr**
Gravatar implemented

### CODE REVIEW

**Comment by gizzmojr** [comment]
The picture hover seems clunky, if not just simply silly. Thoughts?

**Comment by leif81** [comment]
Ya I could live without the hover effect.

**Comment by gizzmojr** [comment]
Ok fixed the user sort problem I caused. Think this is ready to go. I'm seeing <20ms response times for icon queries and the browser is caching between loads. 

Optimization would moving the fetch out of when we do addCards and into a parallel request (loadAssigneesList?) and include an object for name/email/link relation for call too when the card is being made. 

As is, it's barely noticeable on first load of the board doing it the easy way.

**Comment by leif81** [comment]
Hmm I'm seeing a large delay when doing a user filter now. For example, with a full board (prod=Bugzilla milestone=6.0) select a single user. It takes about 3-4 seconds to redraw. Not sure why yet.

File: README.adoc [hunk 1]
```diff
@@ -36,3 +36,6 @@ These are optional settings that can be manually added to the page URL to contro
  ** Disable comment count loading because it's a bit expensive. e.g. `comments=false`
  * `autocomments`
  ** Enable auto comments when making changes to bugs. e.g. `autocomment=true`
+ * `picture`
+ ** Use a picture site option.
+ 1. gravatar
```
Comment (leif81) [inline README.adoc]: I wonder if we are likely to support other sources? I think gravatar is the only (popular) one that takes an email address and returns a photo. We can probably simplify this to `gravatar=true` eh?

Comment (gizzmojr) [reply README.adoc]: Hey, if you're calling it, I'm all for less is more. Gravatar is pretty easy to implement, and I'm very doubtful of using social media api's. 
Google OATH is a tricky beast, each person requesting their own API key and such. Third party solutions are available, http://www.avatarapi.com/

I like how you mentioned adding private work emails into Gravatar works nicely for our use case, and very easy for the individual to set it up.

**Comment by gizzmojr** [comment]
As the cards get more elements, doing a display none forces deletion of all child elements. When you do the filter of assignee and a card is deleted, it forces a reflow of the whole board. On very large boards, this is expensive on render time. 

Chrome Timeline in debug tools will break this down. 
I believe without the img, doing _just_ text, the reflow was very quick. I did some tests with no child img in the span and the result was the same if I recall. 
Thinking on it, I believe it's about more child DOM elements in the cards is the issue. 

In either case, doing a onetime reflow is beneficial in my opinion. 
The committed code in 1f96998fe50c0495db0ceee9661f5f119f77e52f can be improved upon, but paves the way for optimization I think.

**Comment by gizzmojr** [comment]
Can't say I understand why (Chrome at least) innerText is so expensive to parse then innerHTML. As determined, the Timeline shows renderer only doing one reflow (more or less).

Can't argue a simple fix is the best one, my opinion is :confused:

**Comment by gizzmojr** [comment]
Ok tested this on the demo site and it's just as responsive before this PR. Code looks it's cleaned up and I already pushed the gravatar variable use. Let's do it!

**Review by gizzmojr — COMMENTED** [review]
Suppose since we're only supporting gravatar, and we're requesting the image size.
Had it originally for general usage.

**Comment by leif81** [comment]
:+1:

**Pull request merged**